scripts/cert-manager/app-registration-cert-mananger.sh

```diff
@@ -0,0 +1,304 @@
+  "tenantId": "${APP_TENANT}"
+}
+EOF
+
+# Update keyvault with secret
+printf "Updating keyvault \"$AZ_RESOURCE_KEYVAULT\"..."
+if [[ $(az keyvault secret set --name "$APP_REGISTRATION_CERT_MANAGER" --vault-name "$AZ_RESOURCE_KEYVAULT" --file "$UPDATED_SECRET_VALUES_FILE" --expires "$EXPIRATION_DATE" 2>&1) == *"ERROR"* ]]; then
+    az keyvault secret set --name "$APP_REGISTRATION_CERT_MANAGER" --vault-name "$AZ_RESOURCE_KEYVAULT" --file "$UPDATED_SECRET_VALUES_FILE" --expires "$EXPIRATION_DATE"
+    echo -e "\nERROR: Could not update secret in keyvault \"$AZ_RESOURCE_KEYVAULT\". Exiting..."
+    exit 1
+fi
+printf " Done\n"
+
+rm $UPDATED_SECRET_VALUES_FILE
+fi
+
+#######################################################################################
+### Assign custom permission on TXT records
+###
+ROLENAME="DNS TXT Contributor"
+echo ""
+if [[ $USER_PROMPT == true ]]; then
+    while true; do
+        read -p "Create role '$ROLENAME'? (Y/n) " yn
+        case $yn in
+            [Yy]* ) break;;
+            [Nn]* ) echo "Quitting."; exit 0;;
+            * ) echo "Please answer yes or no.";;
+        esac
+    done
+fi
+
+CUSTOMDNSROLE_JSON="cert-mananger-custom-dns-role.json"
+test -f "$CUSTOMDNSROLE_JSON" && rm "$CUSTOMDNSROLE_JSON"
+cat <<EOF >>${CUSTOMDNSROLE_JSON}
+{
+    "Name": "$ROLENAME",
+    "Id": "",
+    "IsCustom": true,
+    "Description": "Can manage DNS TXT records only.",
+    "Actions": [
+        "Microsoft.Network/dnsZones/TXT/*",
+        "Microsoft.Network/dnsZones/read",
+        "Microsoft.Authorization/*/read",
+        "Microsoft.Insights/alertRules/*",
+        "Microsoft.ResourceHealth/availabilityStatuses/read",
+        "Microsoft.Resources/deployments/*",
+        "Microsoft.Resources/subscriptions/resourceGroups/read",
+        "Microsoft.Support/*"
+    ],
+    "NotActions": [
+    ],
+    "AssignableScopes": [
+        "/subscriptions/${AZ_SUBSCRIPTION_ID}/resourceGroups/${AZ_RESOURCE_GROUP_COMMON}/providers/Microsoft.Network/dnszones/${AZ_RESOURCE_DNS}"
+    ]
+}
+EOF
+APP_ID=$(az role definition list --name "$ROLENAME" -o tsv)
+
+if [ ! -z "$APP_ID" ]; then
+echo -e "$ROLENAME exists. Skipping";
+else
+echo -e "Creating $ROLENAME"
+az role definition create --role-definition ./cert-mananger-custom-dns-role.json
+fi
+
+
+#######################################################################################
+### Assign members to role
+###
+
+echo ""
+if [[ $USER_PROMPT == true ]]; then
+    while true; do
+        read -p "Assign '$APP_REGISTRATION_CERT_MANAGER' permission to $ROLENAME? (Y/n) " yn
+        case $yn in
+            [Yy]* ) DNSTXT_PERMISSIONS=true; break;;
+            [Nn]* ) echo "Skipping."; break;;
+            * ) echo "Please answer yes or no.";;
+        esac
+    done
+fi
+
+DEVELOPERS=$(az ad app list --filter "(displayName eq '$APP_REGISTRATION_CERT_MANAGER')" --query "[].appId" --output tsv)
+
+if [[ $DNSTXT_PERMISSIONS == true ]]; then
+printf "Adding permissions to role..."
```

Comment:
here you assign a role to the app registration. example:
```suggestion
printf "Assigning role to app registration..."
```